docker/mcp/app.py

```
@@ -177,35 +177,31 @@ async def event_generator() -> AsyncIterator[Dict[str, Any]]:
-                    
-                logging.debug(f"Using server's SERPER_API_KEY for search")
-            elif not api_key:
-                logging.error("No API key provided in URL path")
-                yield {"event": "error", "data": "API key not provided in URL path."}
+            # Validate against WEBCAT_API_KEY
+            if api_key != WEBCAT_API_KEY:
+                logging.error(f"Invalid WebCAT API key provided: {api_key[:4]}...{api_key[-4:] if len(api_key) > 8 else '****'}")
+                yield {"event": "error", "data": "Authentication failed: Invalid API key."}
+                return
+                
+            logging.debug(f"Using provided API key from URL path")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)